server.js

```diff
@@ -0,0 +1,96 @@
+const net = require("net");
+const files = require("./bodytext.js");
+
+const PORT = 8080;
+
+const server = net.createServer(socket => {
+  socket.on("data", chunk => {
+    let arr = chunk.toString().split("\r\n");
+
+    let requestObj = {};
+
+    arr
+      .filter(function(element) {
+        return element.length !== 0;
+      })
+      .forEach(function(element) {
+        if (element.indexOf(":") !== -1) {
+          let splitElement = element.split(":");
+          requestObj[splitElement[0]] = splitElement[1];
+        } else {
+          let splitElement = element.split(" ");
+          requestObj.URI = splitElement;
+        }
+      });
```

Comment:
This is better but since this goes across the entire request message, you might actually find a situation where the body might have a : in it and might trigger the if statement. What I normally would do is try to figure out how to split the header section from the body section and just run your array functions on just the header section. You don't need to do this but it's something to think about.